### PR TITLE
Hide FormatMail details behind MailAction

### PR DESCRIPTION
**Pull request by Morendil**
Alternative proposal to #84

### CODE REVIEW

File: lib/betagouvbot/format_mail.rb [hunk 1]
```diff
@@ -26,6 +26,10 @@ def call(context)
       end
     end
 
+    def format_recipients(context)
```
Comment (maukoquiroga) [inline lib/betagouvbot/format_mail.rb]: I'd prefer to leave this method private, and change `.call` return as suggested ↑ .

File: lib/betagouvbot/format_mail.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ def initialize(subject, body_t, recipients, sender)
     # @note Email data files consist of 1 subject line plus body
     def self.from_file(body_path, recipients = [], sender = 'secretariat@beta.gouv.fr')
       subject, *rest = File.readlines(body_path)
-      new(subject.strip, rest.join, recipients, sender)
+      new(subject.to_s.strip, rest.join, recipients, sender)
     end
 
     def call(context)
```
Comment (maukoquiroga) [inline lib/betagouvbot/format_mail.rb]: Alternative:

```ruby
def call(context)
  [format_recipients(context), format_message(context)]
end

private

def format_recipients(context)
  # ...
end

def format_message(context)
  {}.tap do |format|
    add_from(context, format)
    add_content(context, format)
    add_personalizations(context, format)
  end
end
```

**Comment by maukoquiroga** [comment]
Like it, check my comments.

File: lib/betagouvbot/format_mail.rb (hunk 1 above)
Comment (Morendil) [inline lib/betagouvbot/format_mail.rb]: It's used by `MailAction`, not just internally.

File: lib/betagouvbot/format_mail.rb (hunk 2 above)
Comment (maukoquiroga) [inline lib/betagouvbot/format_mail.rb]: Or, alternatively, make them public and change class name and remove `.call`:

```ruby
class MailFormatter
  module_method

  class << self
    def format_recipients(context)
       # ...
    end

    def format_message(context)
      # ...
    end

    private

    # ...
  end
end
```

WDYT?

File: lib/betagouvbot/format_mail.rb (hunk 1 above)
Comment (maukoquiroga) [inline lib/betagouvbot/format_mail.rb]: Yep, check my other comment.

File: lib/betagouvbot/format_mail.rb (hunk 2 above)
Comment (maukoquiroga) [inline lib/betagouvbot/format_mail.rb]: Just FTR, what I'd have done in a functional language:

```elixir
defmodule MailFormatter do
  def call(:recipients, context) do: blah
  def call(:message, context) do: blahbhabha
end
```

Which in Ruby we would have to write like:

```ruby
class MailFormatter
  class Base
    # ...
   end

  class Recipients < Base
    def call(context); blah; end
  end

  class Message < Base
    def call(context); blahblah; end
  end
end
```

which is, sadly, too verbose and ugly.

My preferred alternative in Ruby would be:

```ruby
class MailFormatter # note the class name change
  def format_recipients(context); blah; end
  def format_message(context) blahblah; end
end
```

**Pull request merged**